### PR TITLE
CUSTCOM-253 Patch ArrayIndexOutOfBoundsException Defect

### PR DESCRIPTION
**Pull request by MattGill98**
Upstream Issue: eclipse-ee4j/mojarra#4650
Upstream PR: eclipse-ee4j/mojarra#4657

Check cherry-picked commit for actual change

### CODE REVIEW

File: impl/src/main/java/com/sun/faces/renderkit/html_basic/HtmlResponseWriter.java [hunk 1]
```diff
@@ -915,6 +915,8 @@ public void writeText(char text[]) throws IOException {
         }
         closeStartIfNecessary();
 
+        if (text.length == 0) return;
```
Comment (dmatej) [inline impl/src/main/java/com/sun/faces/renderkit/html_basic/HtmlResponseWriter.java:918]: https://rules.sonarsource.com/java/RSPEC-121 (curly braces)

File: impl/src/main/java/com/sun/faces/renderkit/html_basic/HtmlResponseWriter.java [hunk 2]
```diff
@@ -956,6 +958,8 @@ public void writeText(Object text, String componentPropertyName)
         closeStartIfNecessary();
         String textStr = text.toString();
 
+        if (textStr.length() == 0) return;
```
Comment (dmatej) [inline impl/src/main/java/com/sun/faces/renderkit/html_basic/HtmlResponseWriter.java:961]: https://rules.sonarsource.com/java/RSPEC-121 (curly braces)

**Review by dmatej — APPROVED** [review]
Except that formatting the code seems functionaly ok.

File: impl/src/main/java/com/sun/faces/renderkit/html_basic/HtmlResponseWriter.java (hunk 1 above)
Comment (MattGill98) [reply impl/src/main/java/com/sun/faces/renderkit/html_basic/HtmlResponseWriter.java:918]: It's an upstream commit, so I figured it wasn't worth reformatting to be reformatted again next release. Similarly it wouldn't seem worth a unique upstream PR to reformat this

Comment (dmatej) [reply impl/src/main/java/com/sun/faces/renderkit/html_basic/HtmlResponseWriter.java:918]: Ok, than it makes sense to me, I missed links in the description :-)

**Pull request merged**